scijava/scijava-ops-api/src/main/java/org/scijava/ops/api/InfoChainGenerator.java

```diff
@@ -38,23 +38,27 @@ InfoChain generate(String signature, Map<String, OpInfo> idMap,
 	 * @param generators the list of {@link InfoChainGenerator}s
 	 * @return the {@link InfoChainGenerator} best suited to the task
 	 */
-	static Optional<InfoChainGenerator> findSuitableGenerator(String signature,
+	static InfoChainGenerator findSuitableGenerator(String signature,
 		Collection<InfoChainGenerator> generators)
 	{
-		Optional<InfoChainGenerator> gen = generators.stream() //
+		List<InfoChainGenerator> suitableGenerators = generators.stream() //
 			.filter(g -> g.canGenerate(signature)) //
-			.max((i1, i2) -> (int) (i1.priority() - i2.priority()));
-		return gen;
+			.collect(Collectors.toList());
+		if (suitableGenerators.size() == 0) throw new IllegalArgumentException(
+			"No InfoChainGenerator in given collection " + generators +
+				" is able to reify signature " + signature);
+		if (suitableGenerators.size() > 1) throw new IllegalArgumentException(
+			"Signature " + signature +
+				" is able to be reified by multiple generators: " + suitableGenerators);
```

Comment:
If I'm reading this right: you got rid of generator priority, and now assume that only one of the discovered generators will handle each case. If someone writes a plugin that has any overlap with existing generation cases, it will break the system because we'll hit this case of more than one suitable generator.

In this case, I would consider dropping the extensibility. There is an argument to be made to keep it, if new sorts of generation need to be accommodated, and people take care to only handle their newly added cases unambiguously. But if we don't have a clear vision for any new generation scenarios, we may want to simply hardcode the list of generators to the ones we provide, and leave it at that. Could make this code simpler.

Reply:
scijava/scijava#87

Reply:
Resolving this in favor of that issue.